src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -91,6 +92,11 @@ trait Formats
       withServicePorts ++ Json.obj("healthCheckResults" -> task.healthCheckResults)
     else
       withServicePorts
+
+    if (task.mesosTaskUrl.nonEmpty)
+      enrichedJson ++ Json.obj("mesosTaskUrl" -> task.mesosTaskUrl)
```

Comment:
Json.obj should correctly deal with `None` values, so no need for an if-statement.